compiler/luci/service/src/Nodes/CirclePad.cpp

```diff
@@ -35,7 +34,8 @@ loco::TensorShape Algorithm::visit(const luci::CirclePad *node)
 {
   // TODO support non-const case
   auto paddings = loco::must_cast<luci::CircleConst *>(node->paddings());
-  return use_paddings(node, paddings);
+  auto input_shape = circle_shape(loco::must_cast<const luci::CircleNode *>(node->input()));
```

Comment:
There is a possibility that this line might be duplicated later. Below is some methods to remove the duplication. Which one do you think would be better?
```suggestion
  loco::TensorShape pad_shape(const luci::CircleNode *node, const luci::CircleConst *paddings)
  {
  // ...
        auto input_shape = circle_shape(loco::must_cast<const luci::CircleNode *>(node->arg(0)));
 //  ....
  }
  
```

Reply:
Ah, I missed this PR :cry: Sorry for a bit late. 

IMO, duplicating one line is fine. 
Also, for me, the function that gets `TensorShape` and returns `TensorShape` is more intuitive (easy to guess the function role). 

```c++ 
loco::TensorShape pad_shape(const loco::TensorShape &input_shape, const luci::CircleConst *paddings)
```

Reply:
> IMO, duplicating one line is fine. Also, for me, the function that gets `TensorShape` and returns `TensorShape` is more intuitive (easy to guess the function role).
> 

I agree with this.

Reply:
I think you could convert this draft into PR and requests a review to  `@Samsung/one_compiler` members :)